File: Tutorial1-1DaggerBasics/src/main/java/com/example/tutorial1daggerbasics/di/ApplicationComponent.java
package com.example.tutorial1daggerbasics.di;

import com.example.tutorial1daggerbasics.MainActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {ApplicationModule.class})
public interface ApplicationComponent {
    void inject(MainActivity mainActivity);
}

File: Tutorial1-1DaggerBasics/src/main/java/com/example/tutorial1daggerbasics/di/ApplicationComponent.kt
package com.example.tutorial1daggerbasics.di

import com.example.tutorial1daggerbasics.MainActivity
import dagger.Component
import javax.inject.Singleton

/**
 * Component creates graph to inject objects.
 */
@Singleton
@Component(modules = [ApplicationModule::class])
interface ApplicationComponent {

    fun inject(mainActivity: MainActivity)

}